### ISSUE TITLE
Uplift i18n code to use dynamic imports

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#1720]
# Feature Request

## Description
Currently, we use `require.ensure` to dynamically load i18n data based on what the current locale is set as. This is based on the [recommendation from the Intl polyfill we used](https://github.com/andyearnshaw/Intl.js#intljs-and-browserifywebpack). This has worked well, however, webpack recommends using dynamic imports over `require.ensure`. There is a possibility `require.ensure` becomes deprecated, however [that doesn't seem likely yet](https://github.com/webpack/webpack/issues/5909#issuecomment-341988046).
This issue is to start discussion on the update.

**Comment by bjankord** [comment]
There are a couple of PRs/Issues on the Intl polyfill repo for how to switch require.ensure code over to using imports.

* https://github.com/andyearnshaw/Intl.js/pull/299
```diff
-if (!global.Intl) {
-    require.ensure([
-        'intl',
-        'intl/locale-data/jsonp/en.js'
-    ], function (require) {
-        require('intl');	
-        require('intl/locale-data/jsonp/en.js');	
-        runMyApp()	
-    });
+ const selectedLocale = 'en'
+ import('intl').then(() => import(`intl/locale-data/jsonp/${selectedLocale}.js`)).then(runMyApp)
```

* https://github.com/andyearnshaw/Intl.js/issues/300
```js
async function initIntl() {
  if (typeof Intl === 'object') {
    return
  }
  await Promise.all([
    import('intl'),
    import('intl/locale-data/jsonp/en')
  ])
}

initIntl().then(...)
```

**Comment by bjankord** [comment]
Counting this as low priority as the change would be non-passive in that it would require consumers to start using [babel-plugin-syntax-dynamic-import](https://babeljs.io/docs/en/babel-plugin-syntax-dynamic-import/) . Along with this, there was hope that this change would enable use with create-react-app without needing to eject, but there is [an issue in CRA](https://github.com/facebook/create-react-app/issues/4477) where they don't support dynamic imports in imported code, only in app code.

**Comment by bjankord** [comment]
They've added support for dynamic imports in node_modules, (where terra-i18n code would live in CRA app) in the `next` branch of CRA and it will be available in create-react-app v2. Need to test and see how terra-i18n works with this change.

**Comment by noahbenham** [comment]
Leaving this here in case anyone has issues with Create React App 2.

Discussed with @bjankord offline - CRA2 consumers will also need to add `dir="ltr` to the `html` tag of `public/index.html`. Without it, this behavior could be seen:

| <img width="861" alt="screen shot 2018-10-04 at 12 14 32 pm" src="https://user-images.githubusercontent.com/6981195/46491964-dfb3e080-c7d1-11e8-9789-97031d45fb8b.png">  | <img width="860" alt="screen shot 2018-10-04 at 12 14 40 pm" src="https://user-images.githubusercontent.com/6981195/46491965-dfb3e080-c7d1-11e8-912a-f22d64e2b1bb.png"> |
| ------------- | ------------- |

**Comment by bjankord** [comment]
With the work for https://github.com/cerner/terra-core/issues/1941, we'll be removing  [defaultIntlLoaders.js](https://github.com/cerner/terra-core/blob/bab8698ad84b8d1222c85b28588549692e6217c8/packages/terra-i18n/src/defaultIntlLoaders.js), [defaultTranslationsLoaders.js](https://github.com/cerner/terra-core/blob/7daa4d6bad0280659001ff90aa96b7720ff5069d/packages/terra-i18n/src/defaultTranslationsLoaders.js) so that will resolve the require.ensure syntax in those files. We've also [updated terra-toolkit to allow aggregate-translation files to be generated with dynamic imports](https://github.com/cerner/terra-toolkit/issues/175). That just leaves the require.ensure syntax in [i18nLoader.js](https://github.com/cerner/terra-core/blob/master/packages/terra-i18n/src/i18nLoader.js#L21). This code can be reworked so it handles the Intl check in a separate file and exports either the intl polyfill on window.Intl based on intl support removing the need for `require.ensure`.

**Comment by bjankord** [comment]
This has been resolved in terra-base@5.0.0/terra-i18n@4.0.0
